jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -0,0 +1,193 @@
+    */
+    getInitialState: function() {
+      // Setting a default state object.
+      return {
+        autoPlace: this.props.autoPlaceDefault
+      };
+    },
+
+    /**
+      Close the summary.
+
+      @method _close
+    */
+    _close: function() {
+      this.props.changeState({
+        sectionC: {
+          component: null,
+          metadata: {}
+        }
+      });
+    },
+
+    /**
+      Handles calling to clear the ecs and then closing the deployment
+      summary.
+
+      @method _handleClear
+    */
+    _handleClear: function() {
+      this.props.ecsClear();
+      this._close();
+    },
+
+    /**
+      Handle committing when the deploy button in the summary is clicked.
+
+      @method _handleDeploy
+    */
+    _handleDeploy: function() {
+      if (this.state.autoPlace) {
+        this.props.autoPlaceUnits();
+      }
+      // The env is already bound to ecsCommit in app.js.
+      this.props.ecsCommit();
+      this.setState({hasCommits: true}, () => {
+        this._close();
+      });
+    },
+
+    /**
+      Handle navigating to the machine view.
+
+      @method _handleViewMachinesClick
+    */
+    _handleViewMachinesClick: function() {
+      this.props.changeState({
+        sectionB: {
+          component: 'machine',
+          metadata: {}
+        },
+        sectionC: {
+          component: null,
+          metadata: {}
+        }
+      });
+    },
+
+    /**
+      Handle changes to the placement radio buttons.
+
+      @method _handlePlacementChange
+      @param {Object} e The click event.
+    */
+    _handlePlacementChange: function(e) {
+      this.setState({
+        autoPlace: e.currentTarget.getAttribute('data-placement') === 'placed'
+      });
+    },
+
+    /**
+      Generate the list of change items.
+
+      @method _generateChangeItems
+      @returns {Array} The collection of changes.
+    */
+    _generateChangeItems: function() {
+      var changeList = this.props.changeDescriptions;
+      var changes = [];
+      changeList.forEach(function(change, i) {
+        changes.push(
+          <juju.components.DeploymentSummaryChangeItem
+            key={i}
+            change={change} />
+          );
+      }, this);
+      return changes;
+    },
+
+    render: function() {
+      var listHeaderClassName = 'deployment-summary-change-item ' +
+          'deployment-summary__list-header';
+      var buttons = [{
+        title: 'Clear changes',
+        action: this._handleClear
+      }, {
+        title: 'Deploy',
+        action: this._handleDeploy,
+        type: 'confirm'
+      }];
+      return (
+        <div className="deployment-panel__child">
+          <div className="deployment-panel__content">
+            <div className="twelve-col">
+              <div className="inner-wrapper">
+                <h2 className="deployment-panel__title">
+                  Deployment summary
+                </h2>
+                <juju.components.DeploymentSummaryPlacement
```

Comment:
Do we want to set it up so that in sandbox we can still force it to give the option, for testing purposes?